crates/swc_error_reporters/src/error.rs

```diff
@@ -0,0 +1,15 @@
+/// OnlyDiagnosticsError represents a syntax error detected during the SWC
+/// parsing phase.
+///
+/// Although a syntax error is reported by the SWC parser, this error is handled
+/// on the JavaScript side. The Rust side will ignore it during processing.
+#[derive(Debug)]
+pub struct OnlyDiagnosticsError;
+
+impl std::fmt::Display for OnlyDiagnosticsError {
+    fn fmt(&self, f: &mut std::fmt::Formatter<'_>) -> std::fmt::Result {
+        write!(f, "Syntax Error")
```

Comment:
If this is the error message, I think it's better to name is `SyntaxError`, although it can be done in a next PR anyway